### PR TITLE
Perform POM files cleanup.

### PR DESCRIPTION
**Pull request by karelmaxa**
I performed base cleanup of Maven POM files. PR is primarily addressing the following issues:

* Structure of submodule POM files is different.
* Maven POM files contain configuration that is commented out.
* Build of *man-pages* is not working.

### CODE REVIEW

File: opendj-server-legacy/pom.xml [hunk 1]
```diff
@@ -16,1349 +16,1348 @@
...
+                    <execution>
+                        <id>opendj-manifest</id>
+                        <goals>
+                            <goal>manifest</goal>
+                        </goals>
+                        <configuration>
+                            <instructions>
+                                <Bundle-Version>${project.version}</Bundle-Version>
+                                <Export-Package>org.forgerock.opendj.server.embedded</Export-Package>
+                                <Import-Package>
```
Comment (pavelhoral) [inline opendj-server-legacy/pom.xml:576]: This needs to be addressed after JEB is not distributed as a separated artifact. I think we should wait for @fyrbach's work before merging this.

File: pom.xml [hunk 2]
```diff
@@ -462,16 +437,16 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-project-info-reports-plugin</artifactId>
-                <version>2.4</version>
+                <version>3.4.1</version>
```
Comment (pavelhoral) [inline pom.xml:440]: This is inherited from parent. We should avoid plugin definition if not necessary.

Comment (karelmaxa) [reply pom.xml]: OK, version from the parent is the same so i removed it.

**Review by pavelhoral — APPROVED** [review]
LGTM

**Pull request merged**